bricks/strings/rounding.h

```diff
@@ -36,14 +36,64 @@ namespace current {
 namespace strings {
 
 // Rounds the number to have `n_digits` significant digits.
-inline std::string RoundDoubleToString(double value, size_t n_digits) {
+inline std::string RoundDoubleToString(double value, size_t n_digits, bool plus_sign = false) {
   CURRENT_ASSERT(n_digits >= 1);
-  CURRENT_ASSERT(n_digits <= 100);
-  // `value` will be between `10^dim` and `10^(dim+1)`.
-  const int dim = static_cast<int>(std::floor((std::log(value) / std::log(10.0)) + 1e-6));
-  const double k = std::pow(10.0, static_cast<double>(dim - static_cast<int>(n_digits) + 1));
+  CURRENT_ASSERT(n_digits <= 15);
+
+  bool negative = false;
+  if (value < 0) {
+    value = -value;
+    negative = true;
+  }
+
+  if (value < 1e-16) {
+    return "0";
```

Comment:
@dkorolev Does it make sense to return `+0` in case of `plus_sign = true`?

Reply:
Good idea. Shall we wait until if and when it becomes important? :-)